### PR TITLE
Close #7: Set environment variables

### LINKED ISSUE #7: [feature reequest] Set environment variable
Sometimes it needs to set environment variables for executing command. For example: cross compilation:

```
init_cmds = [
    ["GOOS=windows", "GOARCH=amd64", "go", "build", "-o", "target_win.exe", "./src"],
    ["GOOS=linux", "GOARCH=arm", "go", "build", "-o", "target_arm", "./src"],
    ["go", "build", "-o", "target", "./src"],
    ["echo", "answer=42"],
    ["./target"]
]
```

### PR DESCRIPTION
**Pull request by hmgle**
An example of setting env var:

```
[run]
init_cmds = [
    ["GOOS=windows", "GOARCH=amd64", "go", "build", "-o", "foobar.exe", "./foobar.go"],
    ["GOOS=linux", "GOARCH=arm", "go", "build", "-o", "foobar_arm", "./foobar.go"],
    ["go", "build", "-o", "foobar"],
    ["echo", "answer=42"],
    ["./foobar"]
]                                   # Commands run in start
watch_all = true                    # Watch all sub-directories
watch_dirs = []                     # Directories to watch
watch_exts = [".go"]                # Extensions to watch
ignore = [".git", "node_modules"]   # Directories to exclude from watching
ignore_files = []                   # Regexps for ignoring specific notifies
build_delay = 1500                  # Minimal interval to Trigger build event
interrupt_timout = 15               # Time to wait until force kill
cmds = [
    ["GOOS=windows", "GOARCH=amd64", "go", "build", "-o", "foobar.exe", "./foobar.go"],
    ["GOOS=linux", "GOARCH=arm", "go", "build", "-o", "foobar_arm", "./foobar.go"],
    ["go", "build", "-o", "foobar"],
    ["echo", "answer=42"],
    ["./foobar"]
]                                   # Commands to run

[sync]
listen_addr = ":5050"
remote_addr = ":5050"
```

### CODE REVIEW

File: cmd/run.go [hunk 1]
```diff
@@ -75,7 +109,14 @@ func notify(cmds [][]string) {
...
+		env, cmd := splitEnvCmd(args)
+		backupEnv, err := setEnvVar(env)
+		if err != nil {
+			log.Error("Fail to set environment variable %v - %v", env, err)
+			fmt.Print("\x07")
+			return
+		}
 		command := exec.Command(cmd[0], cmd[1:]...)
 		command.Stdout = os.Stdout
 		command.Stderr = os.Stderr
```
Comment (unknwon) [inline cmd/run.go:122]: See [exec.Command.Env](https://gowalker.org/os/exec#Cmd) which allows you set env vars for a specific command without mess up wider range. May save some work of setting/restoring env vars.

**Comment by unknwon** [comment]
Thanks!

Comments are made to improve this PR.

**Comment by hmgle** [comment]
If exec.Command.Env is not nil, it will initialize an environment  just contain command.Env, not inherit the current environment. Adding os.Environ to command.Env is necessary:

``` go
env, cmd := splitEnvCmd(args)
command := exec.Command(cmd[0], cmd[1:]...)
if len(env) > 0 {
    for _, e := range os.Environ() {
        command.Env = append(command.Env, e)
    }
    for _, e := range env {
        command.Env = append(command.Env, e)
    }
}
```

Since length of os.Environ is very long in most cases, I prefer the saving/restoring env vars way.

**Comment by unknwon** [comment]
I see. But it can be optimized:
1. Does not care env vars when no such need, just like the way currently works.
2. Only setting env vars for needed cases and cache that slice, so only first time would be slow.

If you feel confused, leave it to me to implement 🤓

**Comment by hmgle** [comment]
@Unknwon Looking forward for implement. Caching entire command looks good.

**Pull request closed without merging**